core/client/fs/src/main/java/alluxio/client/file/FileSystemContext.java

```diff
@@ -107,106 +96,129 @@
    */
   @GuardedBy("this")
   private WorkerNetAddress mLocalWorker;
-
-  /** The parent user associated with the {@link FileSystemContext}. */
-  private final Subject mParentSubject;
+  private final ClientContext mClientContext;
+  private final String mAppId;
 
   /**
+   * @param alluxioConf Alluxio's configuration
    * @return the instance of file system context with no subject associated
    */
-  public static FileSystemContext get() {
-    return get(null);
+  public static FileSystemContext create(AlluxioConfiguration alluxioConf) {
+    return create(null, alluxioConf);
   }
 
   /**
-   * @param subject the subject associated with this context
-   * @return the instance of the file system context, possibly a shared context
+   * @param subject the parent subject, set to null if not present
+   * @param alluxioConf Alluxio's configuration
+   * @return the context
    */
-  public static FileSystemContext get(Subject subject) {
-    synchronized (CONTEXT_CACHE_LOCK) {
-      FileSystemContext ctx = CONTEXT_CACHE.computeIfAbsent(subject, FileSystemContext::create);
-      ctx.mRefCount++;
-      return ctx;
+  public static FileSystemContext create(Subject subject,
+      @Nullable AlluxioConfiguration alluxioConf) {
+    AlluxioProperties props = null;
+    if (alluxioConf != null) {
+      props = alluxioConf.getProperties();
     }
+    return create(subject, props);
   }
 
   /**
    * @param subject the parent subject, set to null if not present
+   * @param props Alluxio's configuration properties
    * @return the context
    */
-  private static FileSystemContext create(Subject subject) {
-    FileSystemContext context = new FileSystemContext(subject);
-    context.init(MasterInquireClient.Factory.create(), Configuration.global());
+  public static FileSystemContext create(Subject subject, @Nullable AlluxioProperties props) {
```

Comment:
can we have a `FileSystemContext(Subject subject)` constructor so that we don't need to pass `null`?